seaborn/matrix.py

```diff
@@ -964,18 +982,49 @@ def plot_colors(self, xind, yind, **kws):
         if self.row_colors is not None:
             matrix, cmap = self.color_list_to_matrix_and_cmap(
                 self.row_colors, yind, axis=0)
+
+            # Get labels from colors if given as dataframe
+            if isinstance(self.row_colors, pd.DataFrame):
+                xticklabels = self.row_colors.columns
+            elif isinstance(self.row_colors, pd.Series) and \
+                    self.row_colors.name:
+                xticklabels = [self.row_colors.name]
+            else:
+                xticklabels = False
+
             heatmap(matrix, cmap=cmap, cbar=False, ax=self.ax_row_colors,
-                    xticklabels=False, yticklabels=False,
+                    xticklabels=xticklabels, yticklabels=False,
                     **kws)
+
+            # Adjust rotation of labels
+            if xticklabels is not False:
+                self.ax_row_colors.set_xticklabels(
+                    self.ax_row_colors.get_xticklabels(), rotation=90)
```

Comment:
For long labels (and even for labels that are not even that long), they quickly overlap if they are not drawn orthogonally. I think this would be a good default, labels can always be rotated back if this is preferred. Making it configurable might be too much of a hassle.